File: samples/sample.rb
# A small Ruby sample for exercising ruby-mode (tree-sitter).
require "json"

class Greeter
  GREETING = "hello"          # a constant + a string

  def initialize(name)
    @name = name              # an instance variable
    @count = 0
  end

  def greet(loud: false)
    msg = "#{GREETING}, #{@name}!"
    msg = msg.upcase if loud
    @count += 1
    puts msg
    msg
  end
end

g = Greeter.new("world")
3.times { g.greet(loud: true) }

File: samples/sample.ex
# A small Elixir sample for exercising elixir-mode (tree-sitter).
defmodule Greeter do
  @greeting "hello"

  def new(name) do
    %{name: name, count: 0}
  end

  def greet(%{name: name} = state, loud \\ false) do
    msg = "#{@greeting}, #{name}!"
    msg = if loud, do: String.upcase(msg), else: msg
    IO.puts(msg)
    {:ok, %{state | count: state.count + 1}}
  end
end

state = Greeter.new(:world)
Enum.each(1..3, fn _ -> Greeter.greet(state, true) end)
